src/OrchardCore.Modules/OrchardCore.ReverseProxy/Drivers/ReverseProxySettingsDisplayDriver.cs

```diff
@@ -15,63 +16,56 @@ namespace OrchardCore.ReverseProxy.Drivers
 
-        public override IDisplayResult Edit(ReverseProxySettings settings, BuildEditorContext context)
+        public override IDisplayResult Edit(ReverseProxySettings section, BuildEditorContext context)
         {
             return Initialize<ReverseProxySettingsViewModel>("ReverseProxySettings_Edit", model =>
             {
-                if (_memoryCache.Get(RestartPendingCacheKey) != null)
-                    _notifier.Warning(T["The site needs to be restarted for the settings to take effect"]);
-
-                model.EnableXForwardedFor = settings.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedFor);
-                model.EnableXForwardedHost = settings.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedHost);
-                model.EnableXForwardedProto = settings.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedProto);
+                model.EnableXForwardedFor = section.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedFor);
+                model.EnableXForwardedHost = section.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedHost);
+                model.EnableXForwardedProto = section.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedProto);
             }).Location("Content:2").OnGroup(SettingsGroupId);
         }
 
-        public override async Task<IDisplayResult> UpdateAsync(ReverseProxySettings settings, BuildEditorContext context)
+        public override async Task<IDisplayResult> UpdateAsync(ReverseProxySettings section, BuildEditorContext context)
         {
             if (context.GroupId == SettingsGroupId)
             {
                 var model = new ReverseProxySettingsViewModel();
 
                 await context.Updater.TryUpdateModelAsync(model, Prefix);
 
-                settings.ForwardedHeaders = ForwardedHeaders.None;
+                section.ForwardedHeaders = ForwardedHeaders.None;
 
                 if (model.EnableXForwardedFor)
-                    settings.ForwardedHeaders |= ForwardedHeaders.XForwardedFor;
+                    section.ForwardedHeaders |= ForwardedHeaders.XForwardedFor;
 
                 if (model.EnableXForwardedHost)
-                    settings.ForwardedHeaders |= ForwardedHeaders.XForwardedHost;
+                    section.ForwardedHeaders |= ForwardedHeaders.XForwardedHost;
 
                 if (model.EnableXForwardedProto)
-                    settings.ForwardedHeaders |= ForwardedHeaders.XForwardedProto;
+                    section.ForwardedHeaders |= ForwardedHeaders.XForwardedProto;
 
             }
 
-            if (_memoryCache.Get(RestartPendingCacheKey) == null)
-            {
-                var entry = _memoryCache.CreateEntry(RestartPendingCacheKey);
-                _memoryCache.Set(entry.Key, entry, new MemoryCacheEntryOptions { Priority = CacheItemPriority.NeverRemove });
-            }
+            // Reload the tenant to apply the settings
+            await _orchardHost.ReloadShellContextAsync(_currentShellSettings);
 
```

Comment:
Need to wrap this to only reset site if the settings are good
```
                // If the settings are valid, reload the current tenant.
                if (context.Updater.ModelState.IsValid)
                {
                    await _shellHost.ReloadShellContextAsync(_shellSettings);
                }
```

Reply:
Will do